src/client/article.ts

```diff
@@ -451,20 +454,40 @@ function richLinks(): void {
 		});
 }
 
-setup();
-sendTargetingParams();
-ads();
-videos();
-reportNativeElementPositionChanges();
-topics();
-slideshow();
-formatDates();
-footerInit();
-insertEpic();
-callouts();
-renderComments();
-hasSeenCards();
-initAudioAtoms();
-hydrateQuizAtoms();
-localDates();
-richLinks();
+const init = (): void => {
+	const format = pipe2(
+		getFormat(),
+		map((format) => format),
+		withDefault<Format | null>(null),
+	);
+
+	if (format === null) {
+		return;
+	}
+
+	if (format.design === Design.Media) {
+		sendTargetingParams();
+		ads();
+		slideshow();
+	} else {
+		setup();
+		sendTargetingParams();
+		ads();
+		videos();
+		reportNativeElementPositionChanges();
+		topics();
+		slideshow();
+		formatDates();
+		footerInit();
+		insertEpic();
+		callouts();
+		renderComments();
+		hasSeenCards();
+		initAudioAtoms();
+		hydrateQuizAtoms();
+		localDates();
+		richLinks();
+	}
```

Comment:
Do we not want to do these things for media articles?

Reply:
Maybe we do? I was just refactoring it the way it is currently. The `media.ts` currently only calls `ads` and `slideshow`.. maybe it's a legacy thing. Shall we call these for media articles too?

Reply:
I think so. The reason we want to remove `media.js` is because it's going out of date and missing out on all the changes being made to the main client script 🙂.

There's a possible question mark for ads on media articles. How do ads normally work in galleries?

Reply:
I can't get an SSR gallery but in the templates there's only one ad after the 3rd image. Looking at the code here https://github.com/guardian/apps-rendering/blob/815ec3af5c631bc12c1db926caca145f69b0f439/src/ads.tsx#L7, my guess is that we're not handling it the same way in SSR currently and we're gonna have more than one add in the gallery articles. 🤔